trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py

```diff
@@ -159,7 +159,7 @@ def populate_wing_sections(avl_wing,suave_wing):
             ctrl_surf_at_seg = False 
 
             # condition for the presence of control surfaces in segment 
-            if segments[i_segs].control_surfaces:    
+            if getattr(segments[i_segs],'control_surfaces',False):    
```

Comment:
Did you test this syntax? What happens when the segment has control_surfaces?

Reply:
yeah, the test_AVL regression passes. 

when the segment has control_surfaces, population_wing_sections works as intended. The problem was actually the opposite case (when the segment had no control surfaces). The this part AVL suite works similarly to the make_VLM_wings function in that it takes control surfaces from the wings and splits them up among the proper segments. The difference though is that make_VLM_wings works with a custom Data() object, whereas AVL directly uses Segments. So when there is a Segment without control surfaces, the old syntax threw a get_attribute error because I removed control_surfaces from Segments

Reply:
Yes, I understand. It's just funky, because rather than returning a boolean in the case when it has the attribute it actually returns the attribute. Why not not hasattr?

Reply:
Because the segment could conceivably have an empty Data() object, which you would want to return False

Reply:
I could put a bool(   ) around the getattr if that would make what it is doing clearer

Reply:
Since you removed .control_surfaces from the segments object it's extremely unlikely a user would add an empty control surface.

Reply:
True. I'll change it to hasattr then

Reply:
hmm nvm, it's throwing an error for hasattr(). The create_avl_datastructure may be adding empty control_surfaces. Actually yeah that would make sense if it's adding an empty control_surfaces to the root segment b/c of the way populate_control_surfaces works

Reply:
I see